src/main/resources/META-INF/resources/dbt/js/player.js

```diff
@@ -1,171 +1,107 @@
-
-					if (a.type > b.type) {
-						return 1;
-					}
-					if (a.type < b.type) {
-						return -1;
-					}
-					return 0;
-				});
-
-				var playerToHide, playerToShow;
-				var isAudio = currentOption.attr("data-audio") === "true";
-				var htmlEmbed = jQuery(".mir-player");
-
-				if (isAudio) {
-					playerToHide = myPlayerVideo;
-					playerToShow = myPlayerAudio;
-				} else {
-					getThumbs(currentOption, function (sourceArr) {
-						if (sourceArr && sourceArr.length > 0) {
-							myPlayerVideo.poster(sourceArr[0].src);
-						}
-					});
-					playerToShow = myPlayerVideo;
-					playerToHide = myPlayerAudio;
-				}
-
-				hidePlayer(playerToHide);
-
-				$(playerToShow)
-					.data(
-						"fileNodeUrl",
-						webApplicationBaseURL + "rsc/stat/" + currentOption.parent().attr("label") + "/" + currentOption.text() +
-						".css");
-
-				playerToShow.off("play", evtPlay);
-				playerToShow.show();
-				playerToShow.on("play", evtPlay);
-				playerToShow.src(sourceArr);
-			});
-		});
-	});
\ No newline at end of file
+$(document).ready(function () {
+  var videoChooserElement = $("#videoChooser");
+
+  $(".mir-player video").ready(function () {
+    var videoOptions = videoChooserElement.find("option");
+    if (videoOptions.length === 1) {
+      videoChooserElement.closest(".card-header").hide() || videoChooserElement.hide();
+    } else {
+      videoOptions.filter("[data-is-main-doc=true]").first().prop("selected", true);
+    }
+
+    videoChooserElement.change();
+  });
+
+  // get all sources of selected item in a var and give it to player
+  var hidePlayer = function (player) {
+    if (typeof player !== "undefined") {
+      player.hide();
+      player.pause();
+    }
+  };
+
+  var evtPlay = function () {
+    var fnUrl = $(this).data("fileNodeUrl");
+    if (fnUrl) {
+      $.get(fnUrl, function (_data) { });
+    }
+  };
+
+  videoChooserElement.change(function () {
+    // reuse player
+    var myPlayerVideo, myPlayerAudio;
+    var mPlayer;
+    var selectElement = $(this);
+    var currentOption = selectElement.find(":selected");
+
+    if ($(".mir-player video").length > 0) {
+      myPlayerVideo = selectElement.data("playerVideo");
+      mPlayer = selectElement.data("mediaPlayer");
+
+      if (!mPlayer) {
+        if (myPlayerVideo) {
+          var id = myPlayerVideo.id();
+          var parent = $("#" + id).parent();
+          var cls = $("#" + id).attr("class").split(/\s+/);
+
+          myPlayerVideo.dispose();
+
+          var newPlayer = $(document.createElement("video"));
+          newPlayer.attr("id", id);
+          newPlayer.attr("controls", "");
+          newPlayer.attr("poster", "");
+          newPlayer.attr("preload", "metadata");
+
+          for (var i = 0; i < cls.length; i++) {
+            if (cls[i].indexOf("vjs-") !== 0) {
+              newPlayer.addClass(cls[i]);
+            }
+          }
+          parent.append(newPlayer);
+
+          myPlayerVideo = undefined;
+        }
+
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['id' used outside of binding context.](https://app.codacy.com/manual/ThULB/dbt/pullRequest?prid=5557460)